### ISSUE TITLE
HWinfo64 "Shared Memory Support" option now has a 12 hour limit?

### ISSUE THREAD

**Issue by pengoses** [shayne/hwinfo-streamdeck#81]
I have just set up the Stream Deck HWINFO plugin up for the first time. I am using it to display temperatures while in games, and it's great! Thanks to the developer!

However this streamdeck plugin's setup instructions [here](https://github.com/shayne/hwinfo-streamdeck) seem to be written before HWinfo implemented a 12-hour time limit to the "Shared Memory Support" option. 

It sounds like this disable the plugin every 12 hours, is this correct? If so, what is the recommended way to fix this?

**Comment by robinrm** [comment]
i'm using the paid version from hwinfo and there it's not the case. for the free version i don't know ...

**Comment by mauirixxx** [comment]
> I have just set up the Stream Deck HWINFO plugin up for the first time. I am using it to display temperatures while in games, and it's great! Thanks to the developer!
> 
> However this streamdeck plugin's setup instructions [here](https://github.com/shayne/hwinfo-streamdeck) seem to be written before HWinfo implemented a 12-hour time limit to the "Shared Memory Support" option.
> 
> It sounds like this disable the plugin every 12 hours, is this correct? If so, what is the recommended way to fix this?

The recommended way to fix it is to either restart hwinfo, or buy the paid version of hwinfo and never worry about it again.

**Comment by pengoses** [comment]
> The recommended way to fix it is to either restart hwinfo, or buy the paid version of hwinfo and never worry about it again.

Understood, was wondering if that was the only option. $80 is too much for me to pay for this as it's all I'd use it for; though I understand that devs need the support for their work. I'll have to look for another option for showing stats on the SteamDeck.

**Comment by timdonovanuk** [comment]
$89 now for HWinfo. Crazy. Time to find something else.

**Comment by AlexGotcha** [comment]
Yeah that just happened and its pretty sad, been using HWinfo for a long time together with the streamdeck plug-in and now everything got locked behind a pay-wall, can't have anything nice these days...

Does anyone know of any alternative?

**Comment by btastic** [comment]
This plugin could make some use of the workaround that rainmeter is using:

https://docs.rainmeter.net/tips/hwinfo/

Appreciate if the devs would consider this.

**Comment by AriizuCA** [comment]
I just installed the plug-in too and noticed this issue as well.  It costs $129 to get a perpetual license which is ridiculous for someone who just wants to see the performance tab of task manager on their stream deck...

**Comment by Jaule** [comment]
Hey! a workaround is pretty simple. Install the version 6.40! It has all nessasary sensors and it is still free with shared memory support.
take a look here: [https://www.fosshub.com/HWiNFO-old.html](url)

**Comment by pengoses** [comment]
> Hey! a workaround is pretty simple. Install the version 6.40! It has all nessasary sensors and it is still free with shared memory support. take a look here: [https://www.fosshub.com/HWiNFO-old.html](url)

This is what I ended up doing, however it does not support newer hardware. I am not able to see or display gpu temps on my newer card for example, which was one of the main reasons I wanted to use this.

**Comment by 5e** [comment]
> > Hey! a workaround is pretty simple. Install the version 6.40! It has all nessasary sensors and it is still free with shared memory support. take a look here: [https://www.fosshub.com/HWiNFO-old.html](url)
> 
> This is what I ended up doing, however it does not support newer hardware. I am not able to see or display gpu temps on my newer card for example, which was one of the main reasons I wanted to use this.

Hi, I had the exact same problem as you with older version of HWiNFO not detecting my new hardware anymore, and re-enabling the feature every 12 hours was annoying

(Sorry for the shameless self plug) So I decided to create a new HWiNFO plugin which does not require using 'Shared Memory Support', and instead uses the new 'HWiNFO Gadgets' feature like Rainmeter.  It is fully open source and available to download from my repo, and it also has been submitted to the Stream Deck store but it may take a few days to show up on there

I would have considered creating a pull request to refactor this plugin, however, that is a large task for me as the codebase is a little complex, and I don't have much knowledge in the Go language.

**Comment by Smokex365** [comment]
I just made a powershell script that checks for the process uptime and restarts it if it's been running for more than 4 hours. It's launched my a scheduled task on login or boot and just runs in the background. It's worked well for me to bypass the 12-hour limit.

``` powershell
# Checks for Administrator privileges and opens an elevated prompt is user has Administrator rights
If (-NOT ([Security.Principal.WindowsPrincipal][Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole] "Administrator"))
{   
	$arguments = "& '" + $myinvocation.mycommand.definition + "'"
	Start-Process powershell -Verb runAs -ArgumentList $arguments -WindowStyle Hidden
	Break
}

$hwinfo = "HWiNFO64"
$hwinfopath = "C:\Program Files\HWiNFO64\HWiNFO64.EXE"
$hwworkdir = "C:\Program Files\HWiNFO64\"

#$process = Get-Process -ProcessName $hwinfo
Stop-Process -Name $hwinfo

for(;;){
    If (!(Get-Process -ProcessName $hwinfo -ErrorAction SilentlyContinue))
        {Start-Process -filepath $hwinfopath -WorkingDirectory $hwworkdir}
        $proc = Get-Process -ProcessName $hwinfo | Sort-Object -Property ProcessName -Unique -ErrorAction SilentlyContinue
    If (!$proc -or ($proc.Responding -eq $false) -or ($proc.StartTime -lt (Get-Date).AddHours(-4))) {
        $proc.Kill()
        Start-Sleep -s 5
        Start-Process -filepath $hwinfopath -WorkingDirectory $hwworkdir}
    Start-sleep -s 900
}
```

**Comment by LVNeptune** [comment]
> I just made a powershell script that checks for the process uptime and restarts it if it's been running for more than 4 hours. It's launched my a scheduled task on login or boot and just runs in the background. It's worked well for me to bypass the 12-hour limit.
> 
> ```powershell
> # Checks for Administrator privileges and opens an elevated prompt is user has Administrator rights
> If (-NOT ([Security.Principal.WindowsPrincipal][Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole] "Administrator"))
> {   
> 	$arguments = "& '" + $myinvocation.mycommand.definition + "'"
> 	Start-Process powershell -Verb runAs -ArgumentList $arguments -WindowStyle Hidden
> 	Break
> }
> 
> $hwinfo = "HWiNFO64"
> $hwinfopath = "C:\Program Files\HWiNFO64\HWiNFO64.EXE"
> $hwworkdir = "C:\Program Files\HWiNFO64\"
> 
> #$process = Get-Process -ProcessName $hwinfo
> Stop-Process -Name $hwinfo
> 
> for(;;){
>     If (!(Get-Process -ProcessName $hwinfo -ErrorAction SilentlyContinue))
>         {Start-Process -filepath $hwinfopath -WorkingDirectory $hwworkdir}
>         $proc = Get-Process -ProcessName $hwinfo | Sort-Object -Property ProcessName -Unique -ErrorAction SilentlyContinue
>     If (!$proc -or ($proc.Responding -eq $false) -or ($proc.StartTime -lt (Get-Date).AddHours(-4))) {
>         $proc.Kill()
>         Start-Sleep -s 5
>         Start-Process -filepath $hwinfopath -WorkingDirectory $hwworkdir}
>     Start-sleep -s 900
> }
> ```

I assume by now you have figured this out but it's 12 hours cumulative, not 12 hours in a single run. You can restart it as many times as you want but once the run time exceeds 12 hours you have to re-enable it.

**Comment by ChriStrongL** [comment]
Is it possible to make it compatible with HWINFO32? As far as I know 32bit version does not have this 12-hour limit.

**Comment by Smokex365** [comment]
> > I just made a powershell script that checks for the process uptime and restarts it if it's been running for more than 4 hours. It's launched my a scheduled task on login or boot and just runs in the background. It's worked well for me to bypass the 12-hour limit.
> > ```powershell
> > # Checks for Administrator privileges and opens an elevated prompt is user has Administrator rights
> > If (-NOT ([Security.Principal.WindowsPrincipal][Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole] "Administrator"))
> > {   
> > 	$arguments = "& '" + $myinvocation.mycommand.definition + "'"
> > 	Start-Process powershell -Verb runAs -ArgumentList $arguments -WindowStyle Hidden
> > 	Break
> > }
> > 
> > $hwinfo = "HWiNFO64"
> > $hwinfopath = "C:\Program Files\HWiNFO64\HWiNFO64.EXE"
> > $hwworkdir = "C:\Program Files\HWiNFO64\"
> > 
> > #$process = Get-Process -ProcessName $hwinfo
> > Stop-Process -Name $hwinfo
> > 
> > for(;;){
> >     If (!(Get-Process -ProcessName $hwinfo -ErrorAction SilentlyContinue))
> >         {Start-Process -filepath $hwinfopath -WorkingDirectory $hwworkdir}
> >         $proc = Get-Process -ProcessName $hwinfo | Sort-Object -Property ProcessName -Unique -ErrorAction SilentlyContinue
> >     If (!$proc -or ($proc.Responding -eq $false) -or ($proc.StartTime -lt (Get-Date).AddHours(-4))) {
> >         $proc.Kill()
> >         Start-Sleep -s 5
> >         Start-Process -filepath $hwinfopath -WorkingDirectory $hwworkdir}
> >     Start-sleep -s 900
> > }
> > ```
> 
> I assume by now you have figured this out but it's 12 hours cumulative, not 12 hours in a single run. You can restart it as many times as you want but once the run time exceeds 12 hours you have to re-enable it.

I don't experience any issue with it though I've additionally set the settings file as read-only after configuring it. It works fine for me :tm:.

**Comment by moeilijk** [comment]
The 12 hour limit on Shared Memory Support is HWiNFO's free version policy, a Pro license removes it. Not something a plugin can work around. The plugin now shows a clear unavailable state when the feed stops.

This project is under new maintenance. The plugin core has been completely replaced since this was filed: new engine, new configuration UI, rewritten HWiNFO bridge. The code this was reported against is gone, so we are closing the pre-v3 backlog and starting clean. If you still see this on v3.0, please open a new issue.
